### ISSUE TITLE
Support for unencrypted protocol version

### ISSUE THREAD

**Issue by JohannesSchilling** [neverpanic/co2-fe2o3#4]
I found out i have a device that doesn't do the encrypted encoding. Tested this by replacing [sensor.rs line 84](https://github.com/neverpanic/co2-fe2o3/blob/master/src/sensor.rs#L84) with `let decrypted = buf;`.

I'd be happy to open a PR adding proper support, but wanted to first ask: would you prefer a feature flag (since which hardware i have is pretty much a static thing, no need to be runtime-configurable i guess) or a parameter to `Sensors::new`/new method `Sensors::new_nocrypt`? would have to be threaded through the `Sensors::sensors` detector method, but that would make it easy to just pass a command line argument `--nocrypt` to `Sensors::sensors` as flag for which `new_` method to use. Or add something to the config? but how to properly differentiate different sensors, i.e. would i actually be able to write a config for one proper and one nocrypt sensor, given they have the same USB ids etc.?

in any case, happy to hear your input.

**Comment by JohannesSchilling** [comment]
hm, or maybe we can detect which type we have in `new`.. while it's the same vendor/product id, [this hackaday article on reverse-engineering the protocol](https://hackaday.io/project/5301/logs?sort=oldest) has these entries in the device descriptor (`lsusb -vd 04d9:a052`)
```
  bcdDevice            1.00
  iManufacturer           1 Holtek
  iProduct                2 USB-zyTemp
  iSerial                 3 1.40
```

while mine has
```
  bcdDevice            2.00
  iManufacturer           1 Holtek
  iProduct                2 USB-zyTemp
  iSerial                 3 2.00
```
could you (and/or others, @dnnr maybe?) confirm that you all have the `bcdDevice 1.00`/`iSerial 1.40` version, i.e. it's likely this difference actually corresponds to the different behavior?

**Comment by neverpanic** [comment]
I have two data points to contribute, both work with this code:

```
$ usbconfig -d ugen0.4 dump_device_desc
ugen0.4: <Holtek USB-zyTemp> at usbus0, cfg=0 md=HOST spd=LOW (1.5Mbps) pwr=ON (100mA)

  bLength = 0x0012
  bDescriptorType = 0x0001
  bcdUSB = 0x0110
  bDeviceClass = 0x0000  <Probed by interface class>
  bDeviceSubClass = 0x0000
  bDeviceProtocol = 0x0000
  bMaxPacketSize0 = 0x0008
  idVendor = 0x04d9
  idProduct = 0xa052
  bcdDevice = 0x0100
  iManufacturer = 0x0001  <Holtek>
  iProduct = 0x0002  <USB-zyTemp>
  iSerialNumber = 0x0003  <1.40>
  bNumConfigurations = 0x0001
```

```
$ lsusb -vd 04d9:a052
  bcdDevice            1.00
  iManufacturer           1 Holtek
  iProduct                2 USB-zyTemp
  iSerial                 3 1.40
```

I'm not aware of anybody running multiple sensors on a single instance of this tool, even though the code allows it. It would be nice if we could preserve support for doing this, ideally with both the encrypted and the unencrypted protocol.

I'd suggest let's try the version number first, and if that doesn't work, maybe we could just try whether the data is unencrypted using some other method, e.g. some kind of checksum in the protocol?

**Comment by JohannesSchilling** [comment]
alright, opened #5, will have to rework if anyone with current `master` working on a `2.00` device shows up.

**Comment by neverpanic** [comment]
Fixed by #5.
